packages/maskbook/src/plugins/GoodGhosting/hooks/useGameInfo.ts

```diff
@@ -3,27 +3,31 @@ import { useMemo } from 'react'
 import { useAsyncRetry } from 'react-use'
 import type { AsyncStateRetry } from 'react-use/lib/useAsyncRetry'
 import { useGoodGhostingContract } from '../contracts/useGoodGhostingContract'
-import type { GoodGhostingInfo, Player, TimelineEvent } from '../types'
+import type { GameMetaData, GoodGhostingInfo, Player, TimelineEvent } from '../types'
 import { ZERO_ADDRESS } from '../constants'
 import { useI18N } from '../../../utils'
 import addSeconds from 'date-fns/addSeconds'
 
-export function useGameContractAddress() {
+export function useGameContractAddress(id: string) {
     const { GOOD_GHOSTING_CONTRACT_ADDRESS_FILE } = useGoodGhostingConstants()
 
-    const asyncResult = useAsyncRetry(async (): Promise<{ contractAddress?: string }> => {
-        if (!GOOD_GHOSTING_CONTRACT_ADDRESS_FILE) return {}
+    const asyncResult = useAsyncRetry(async (): Promise<GameMetaData> => {
+        if (!GOOD_GHOSTING_CONTRACT_ADDRESS_FILE)
+            return {
+                contractAddress: '',
+            }
 
         const response = await fetch(GOOD_GHOSTING_CONTRACT_ADDRESS_FILE)
         const data = await response.text()
-        return data ? JSON.parse(data) : {}
+        const gameData = data ? JSON.parse(data) : {}
+        return gameData[id] || gameData.default || {}
     }, [GOOD_GHOSTING_CONTRACT_ADDRESS_FILE])
```

Comment:
You should add `id` to the dependencies.

[id, GOOD_GHOSTING_CONTRACT_ADDRESS_FILE]